# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1521751657 -3600
#      Thu Mar 22 21:47:37 2018 +0100
# Node ID 47f25e164ab14fc125dbf8d5db12f2a1e3d946a9
# Parent  55480087f125d2fac049cd1fc8f0881fae49af7a
Delete graphics properties or allowed values scheduled for deprecation in 4.6

* NEWS: Announce removal of graphics properties/values

* graphics.in.h (axes::properties::x(y)axislocation): Remove "zero" from
  allowed values.
  (patch::properties::normalmode, surface::properties::normalmode): Remove
  properties and associated set/get methods.

* deprecate-props.tst: Remove tests for 4.6. Fix broken test function.

* __gnuplot_draw_axes__.m: Don't check "zero" value for "x(y)axislocation".

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -1,8 +1,21 @@
+Summary of important user-visible changes for version 4.6 (yyyy-mm-dd):
+----------------------------------------------------------------------
+
+ ** The following properties or allowed corresponding values were deprecated in
+    Octave 4.2 and have been removed from Octave 4.6:
+
+      Object               | Property                | Value
+      ---------------------|-------------------------|-------------------
+      axes                 | xaxislocation           | "zero"
+                           | yaxislocation           | "zero"
+      patch                | normalmode              |
+      surface              | normalmode              |
+
 Summary of important user-visible changes for version 4.4 (yyyy-mm-dd):
 ----------------------------------------------------------------------
 
  ** A graphical Variable Editor has been added to the GUI interface.
     It uses a spreadsheet-like interface for quick, intuitive editing
     of variables.  The Variable Editor is launched by double-clicking
     on a variable name in the Workspace Window or by typing
     "openvar VARIABLE_NAME" in the Command Window.
diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -3656,36 +3656,34 @@ public:
       array_property ticklength u , default_axes_ticklength ()
       array_property tightinset r , Matrix (1, 4, 0.0)
       handle_property title SOf , gh_manager::make_graphics_handle ("text", __myhandle__, false, false, false)
       double_property titlefontsizemultiplier u , 1.1
       radio_property titlefontweight u , "{bold}|normal"
       // FIXME: uicontextmenu should be moved here.
       radio_property units SU , "{normalized}|inches|centimeters|points|pixels|characters"
       array_property view u , default_axes_view ()
-      // FIXME: Remove "zero" in 4.6
-      radio_property xaxislocation u , "{bottom}|top|origin|zero"
+      radio_property xaxislocation u , "{bottom}|top|origin"
       color_property xcolor mu , color_values (0.15, 0.15, 0.15)
       radio_property xcolormode , "{auto}|manual"
       radio_property xdir u , "{normal}|reverse"
       bool_property xgrid , "off"
       handle_property xlabel SOf , gh_manager::make_graphics_handle ("text", __myhandle__, false, false, false)
       row_vector_property xlim mu , default_lim ()
       radio_property xlimmode al , "{auto}|manual"
       bool_property xminorgrid , "off"
       bool_property xminortick , "off"
       radio_property xscale alu , "{linear}|log"
       row_vector_property xtick mu , default_axes_tick ()
       // FIXME: should be kind of string array.
       any_property xticklabel S , ""
       radio_property xticklabelmode u , "{auto}|manual"
       double_property xticklabelrotation , 0.0
       radio_property xtickmode u , "{auto}|manual"
-      // FIXME: Remove "zero" in 4.6
-      radio_property yaxislocation u , "{left}|right|origin|zero"
+      radio_property yaxislocation u , "{left}|right|origin"
       color_property ycolor mu , color_values (0.15, 0.15, 0.15)
       radio_property ycolormode , "{auto}|manual"
       radio_property ydir u , "{normal}|reverse"
       bool_property ygrid , "off"
       handle_property ylabel SOf , gh_manager::make_graphics_handle ("text", __myhandle__, false, false, false)
       row_vector_property ylim mu , default_lim ()
       radio_property ylimmode al , "{auto}|manual"
       bool_property yminorgrid , "off"
@@ -3817,21 +3815,16 @@ public:
                          yaxislocation_is ("origin"),
                          xscale.is ("log") ? 2 :
                            (xaxislocation_is ("origin") ? 0 :
                              (xaxislocation_is ("bottom") ? -1 : 1)),
                          ylim);
     }
     void update_yaxislocation (void)
     {
-      // FIXME: Remove warning with "zero" in 4.6
-      if (yaxislocation_is ("zero"))
-        warning_with_id ("Octave:deprecated-property",
-                         "Setting 'yaxislocation' to 'zero' is deprecated, "
-                         "set to 'origin' instead.");
       sync_positions ();
       update_axes_layout ();
       if (xticklabelmode.is ("auto"))
         calc_ticklabels (xtick, xticklabel, xscale.is ("log"),
                          xaxislocation_is ("origin"),
                          yscale.is ("log") ? 2 :
                            (yaxislocation_is ("origin") ? 0 :
                              (yaxislocation_is ("left") ? -1 : 1)),
@@ -3842,21 +3835,16 @@ public:
                          xscale.is ("log") ? 2 :
                            (xaxislocation_is ("origin") ? 0 :
                              (xaxislocation_is ("bottom") ? -1 : 1)),
                          ylim);
       update_ylabel_position ();
     }
     void update_xaxislocation (void)
     {
-      // FIXME: Remove warning with "zero" in 4.6
-      if (xaxislocation_is ("zero"))
-        warning_with_id ("Octave:deprecated-property",
-                         "Setting 'xaxislocation' to 'zero' is deprecated, "
-                         "set to 'origin' instead.");
       sync_positions ();
       update_axes_layout ();
       if (xticklabelmode.is ("auto"))
         calc_ticklabels (xtick, xticklabel, xscale.is ("log"),
                          xaxislocation_is ("origin"),
                          yscale.is ("log") ? 2 :
                            (yaxislocation_is ("origin") ? 0 :
                              (yaxislocation_is ("left") ? -1 : 1)),
@@ -4872,17 +4860,16 @@ public:
       // FIXME: Remove interpreter property in version 4.8
       radio_property interpreter hd , "{tex}|none|latex"
       radio_property linestyle , "{-}|--|:|-.|none"
       double_property linewidth , 0.5
       radio_property marker , "{none}|+|o|*|.|x|s|square|d|diamond|^|v|>|<|p|pentagram|h|hexagram"
       color_property markeredgecolor , color_property (radio_values ("none|{auto}|flat"), color_values (0, 0, 0))
       color_property markerfacecolor , color_property (radio_values ("{none}|auto|flat"), color_values (0, 0, 0))
       double_property markersize , 6
-      radio_property normalmode hsg , "{auto}|manual"
       double_property specularcolorreflectance , 1.0
       double_property specularexponent , 10.0
       double_property specularstrength , 0.9
       array_property vertexnormals m , Matrix ()
       radio_property vertexnormalsmode , "{auto}|manual"
       array_property vertices u , default_patch_vertices ()
       array_property xdata u , default_patch_xdata ()
       array_property ydata u , default_patch_ydata ()
@@ -4989,32 +4976,16 @@ public:
 
       if (cdatamapping_is ("scaled"))
         set_clim (cdata.get_limits ());
       else
         clim = cdata.get_limits ();
     }
 
     void update_data (void);
-
-    void set_normalmode (const octave_value& val)
-    {
-      warning_with_id ("Octave:deprecated-property",
-        "patch: Property 'normalmode' is deprecated and will be removed "
-        "from a future version of Octave.  Use 'vertexnormalsmode' instead.");
-      set_vertexnormalsmode (val);
-    }
-
-    std::string get_normalmode (void) const
-    {
-      warning_with_id ("Octave:deprecated-property",
-        "patch: Property 'normalmode' is deprecated and will be removed "
-        "from a future version of Octave.  Use 'vertexnormalsmode' instead.");
-      return vertexnormalsmode.current_value ();
-    }
   };
 
 private:
   properties xproperties;
 
 public:
   patch (const graphics_handle& mh, const graphics_handle& p)
     : base_graphics_object (), xproperties (mh, p)
@@ -5084,17 +5055,16 @@ public:
       radio_property interpreter hd , "{tex}|none|latex"
       radio_property linestyle , "{-}|--|:|-.|none"
       double_property linewidth , 0.5
       radio_property marker , "{none}|+|o|*|.|x|s|square|d|diamond|^|v|>|<|p|pentagram|h|hexagram"
       color_property markeredgecolor , color_property (radio_values ("none|{auto}|flat"), color_values (0, 0, 0))
       color_property markerfacecolor , color_property (radio_values ("{none}|auto|flat"), color_values (0, 0, 0))
       double_property markersize , 6
       radio_property meshstyle , "{both}|row|column"
-      radio_property normalmode hsg , "{auto}|manual"
       double_property specularcolorreflectance , 1
       double_property specularexponent , 10
       double_property specularstrength , 0.9
       array_property vertexnormals m , Matrix ()
       radio_property vertexnormalsmode u , "{auto}|manual"
       array_property xdata u , default_surface_xdata ()
       string_property xdatasource , ""
       array_property ydata u , default_surface_ydata ()
@@ -5180,32 +5150,16 @@ public:
       update_vertex_normals ();
       set_zlim (zdata.get_limits ());
     }
 
     void update_vertex_normals (void);
 
     void update_vertexnormalsmode (void)
     { update_vertex_normals (); }
-
-    void set_normalmode (const octave_value& val)
-    {
-      warning_with_id ("Octave:deprecated-property",
-        "surface: Property 'normalmode' is deprecated and will be removed "
-        "from a future version of Octave.  Use 'vertexnormalsmode' instead.");
-      set_vertexnormalsmode (val);
-    }
-
-    std::string get_normalmode (void) const
-    {
-      warning_with_id ("Octave:deprecated-property",
-        "surface: Property 'normalmode' is deprecated and will be removed "
-        "from a future version of Octave.  Use 'vertexnormalsmode' instead.");
-      return vertexnormalsmode.current_value ();
-    }
   };
 
 private:
   properties xproperties;
 
 public:
   surface (const graphics_handle& mh, const graphics_handle& p)
     : base_graphics_object (), xproperties (mh, p)
diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -242,27 +242,27 @@ function __gnuplot_draw_axes__ (h, plot_
   endif
 
   if (strcmp (axis_obj.xaxislocation, "top"))
     xaxisloc = "x2";
     xaxisloc_using = "x2";
   else
     xaxisloc = "x";
     xaxisloc_using = "x1";
-    if (any (strcmp (axis_obj.xaxislocation, {"origin", "zero"}))) # FIXME: Remove "zero" in 4.6
+    if (strcmp (axis_obj.xaxislocation, "origin"))
       fputs (plot_stream, "set xzeroaxis;\n");
     endif
   endif
   if (strcmp (axis_obj.yaxislocation, "right"))
     yaxisloc = "y2";
     yaxisloc_using = "y2";
   else
     yaxisloc = "y";
     yaxisloc_using = "y1";
-    if (any (strcmp (axis_obj.yaxislocation, {"origin", "zero"}))) # FIXME: Remove "zero" in 4.6
+    if (strcmp (axis_obj.yaxislocation, "origin"))
       fputs (plot_stream, "set yzeroaxis;\n");
     endif
   endif
 
   have_major_grid = false;
   have_minor_grid = false;
   visible_gls = ! strcmp (axis_obj.gridlinestyle, "none") ...
                 && ! strcmp (axis_obj.gridcolor, "none");
@@ -1491,23 +1491,23 @@ function __gnuplot_draw_axes__ (h, plot_
     else
       if (nd == 3)
         do_border_tick_3d (axis_obj, plot_stream);
       else
         axis_idx = do_border_2d (axis_obj, plot_stream, axis_idx);
         if (isempty (axis_obj.xtick))
         elseif (strcmp (axis_obj.xaxislocation, "top"))
           fprintf (plot_stream, "set x2tics %s nomirror\n", axis_obj.tickdir);
-        else # xaxislocation == "bottom", "origin" or "zero"
+        else # xaxislocation == "bottom" or "origin"
           fprintf (plot_stream, "set xtics %s nomirror\n", axis_obj.tickdir);
         endif
         if (isempty (axis_obj.ytick))
         elseif (strcmp (axis_obj.yaxislocation, "right"))
           fprintf (plot_stream, "set y2tics %s nomirror\n", axis_obj.tickdir);
-        else # yaxislocation == "left", "origin" or "zero"
+        else # yaxislocation == "left" or "origin"
           fprintf (plot_stream, "set ytics %s nomirror\n",  axis_obj.tickdir);
         endif
       endif
     endif
   endif
 
   if (! have_major_grid && ! have_minor_grid)
     fputs (plot_stream, "unset grid;\n");
@@ -1802,20 +1802,20 @@ function idx = do_border_2d (obj, plot_s
   endif
   if (strcmp (obj.box, "on") || strcmp (obj.yaxislocation, "left"))
     arrow (3, obj.ycolor, obj.linewidth, [0,0,0], [0,1,0]);
   endif
   if (strcmp (obj.box, "on") || strcmp (obj.yaxislocation, "right"))
     arrow (4, obj.ycolor, obj.linewidth, [1,0,0], [1,1,0]);
   endif
 
-  if (any (strcmp (obj.xaxislocation, {"origin", "zero"}))) # FIXME: Remove "zero" in 4.6
+  if (strcmp (obj.xaxislocation, "origin"))
     idx = zeroaxis (idx, obj.xcolor, "x");
   endif
-  if (any (strcmp (obj.yaxislocation, {"origin", "zero"}))) # FIXME: Remove "zero" in 4.6
+  if (strcmp (obj.yaxislocation, "origin"))
     idx = zeroaxis (idx, obj.ycolor, "y");
   endif
 
   function idx = zeroaxis (idx, lc, ax)
     idx = idx + 1;
     [~, ltidx] = do_linestyle_command (obj, lc, idx, plot_stream);
     fprintf (plot_stream, "set %szeroaxis ls %d ", ax, ltidx);
     fprintf (plot_stream, "lw %.3f\n", obj.linewidth);
@@ -2171,17 +2171,17 @@ function do_tics (obj, plot_stream, ymir
     do_tics_1 (obj.xtickmode, obj.xtick, obj.xminortick, obj.xticklabelmode,
                obj.xticklabel, obj.xcolor, "x2", plot_stream, true,
                "border", obj.tickdir, ticklength, fontname, fontspec,
                obj.ticklabelinterpreter, obj.xscale, obj.xsgn, gnuplot_term);
     do_tics_1 ("manual", [], "off", obj.xticklabelmode, obj.xticklabel,
                obj.xcolor, "x", plot_stream, true, "border",
                "", "", fontname, fontspec, obj.ticklabelinterpreter,
                obj.xscale, obj.xsgn, gnuplot_term);
-  elseif (any (strcmp (obj.xaxislocation, {"origin", "zero"}))) # FIXME: Remove "zero" in 4.6
+  elseif (strcmp (obj.xaxislocation, "origin"))
     do_tics_1 (obj.xtickmode, obj.xtick, obj.xminortick, obj.xticklabelmode,
                obj.xticklabel, obj.xcolor, "x", plot_stream, true,
                "axis", obj.tickdir, ticklength, fontname, fontspec,
                obj.ticklabelinterpreter, obj.xscale, obj.xsgn, gnuplot_term);
     do_tics_1 ("manual", [], "off", obj.xticklabelmode, obj.xticklabel,
                obj.xcolor, "x2", plot_stream, true, "axis",
                "", "", fontname, fontspec, obj.ticklabelinterpreter,
                obj.xscale, obj.xsgn, gnuplot_term);
@@ -2199,17 +2199,17 @@ function do_tics (obj, plot_stream, ymir
     do_tics_1 (obj.ytickmode, obj.ytick, obj.yminortick, obj.yticklabelmode,
                obj.yticklabel, obj.ycolor, "y2", plot_stream, ymirror,
                "border", obj.tickdir, ticklength, fontname, fontspec,
                obj.ticklabelinterpreter, obj.yscale, obj.ysgn, gnuplot_term);
     do_tics_1 ("manual", [], "off", obj.yticklabelmode, obj.yticklabel,
                obj.ycolor, "y", plot_stream, ymirror, "border",
                "", "", fontname, fontspec, obj.ticklabelinterpreter,
                obj.yscale, obj.ysgn, gnuplot_term);
-  elseif (any (strcmp (obj.yaxislocation, {"origin", "zero"}))) # FIXME: Remove "zero" in 4.6
+  elseif (strcmp (obj.yaxislocation, "origin"))
     do_tics_1 (obj.ytickmode, obj.ytick, obj.yminortick, obj.yticklabelmode,
                obj.yticklabel, obj.ycolor, "y", plot_stream, ymirror,
                "axis", obj.tickdir, ticklength, fontname, fontspec,
                obj.ticklabelinterpreter, obj.yscale, obj.ysgn, gnuplot_term);
     do_tics_1 ("manual", [], "off", obj.yticklabelmode, obj.yticklabel,
                obj.ycolor, "y2", plot_stream, ymirror, "axis",
                "", "", fontname, fontspec, obj.ticklabelinterpreter,
                obj.yscale, obj.ysgn, gnuplot_term);
diff --git a/test/deprecate-props.tst b/test/deprecate-props.tst
--- a/test/deprecate-props.tst
+++ b/test/deprecate-props.tst
@@ -22,52 +22,23 @@
 
 %!function testprop (h, prop, vrs, val = [])
 %!  vrsmax = strsplit (vrs, ".");
 %!  vrscur = strsplit (version (), ".");
 %!  if (num2str (vrsmax{1}) < num2str (vrscur{1}) ||
 %!      num2str (vrsmax{2}) < num2str (vrscur{2}))
 %!    if (isempty (val) && isprop (h, prop))
 %!      error ("Please remove %s property %s", get (h, "type"), prop);
-%!    elseif (! isempty (val) && ! isempty (strfind (val, set (h, prop))))
+%!    elseif (! isempty (val) && any (strcmp (val, set (h, prop))))
 %!      error ("Please remove '%s' from allowed values for %s property %s",
 %!             val, get (h, "type"), prop);
 %!    endif
 %!  endif
 %!endfunction
 
-## patch/surface "normalmode" deprecated in 4.2, remove from 4.5.+
-%!test
-%! hf = figure ("visible", "off");
-%! unwind_protect
-%!   hp = patch ();
-%!   testprop (hp, "normalmode", "4.4");
-%! unwind_protect_cleanup
-%!   close (hf);
-%! end_unwind_protect
-
-%! hf = figure ("visible", "off");
-%! unwind_protect
-%!   hs = surface ();
-%!   testprop (hs, "normalmode", "4.4");
-%! unwind_protect_cleanup
-%!   close (hf);
-%! end_unwind_protect
-
-## axes, "zero" value for "x/yaxislocation" deprecated in 4.2, remove from 4.5.+
-%!test
-%! hf = figure ("visible", "off");
-%! unwind_protect
-%!   ha = axes ();
-%!   testprop (ha, "xaxislocation", "4.4", "zero");
-%!   testprop (ha, "yaxislocation", "4.4", "zero");
-%! unwind_protect_cleanup
-%!   close (hf);
-%! end_unwind_protect
-
 ## annotation rectangle "edgecolor" deprecated in 4.4, remove from 4.7.+
 %!test
 %! hf = figure ("visible", "off");
 %! unwind_protect
 %!   ha = annotation ("rectangle");
 %!   testprop (ha, "edgecolor", "4.6");
 %! unwind_protect_cleanup
 %!   close (hf);
